include/deal.II/lac/la_parallel_block_vector.h

```diff
@@ -348,10 +348,24 @@ namespace LinearAlgebra
        * After calling this method, read access to ghost elements of the
        * vector is forbidden and an exception is thrown. Only write access to
        * ghost elements is allowed in this state.
+       *
+       * @deprecated Use zero_out_ghost_values() instead.
        */
-      void
+      DEAL_II_DEPRECATED void
```

Comment:
I agree with the general PR and with `@deprecated` in the doxygen comment, but I am very unhappy about adding the `DEAL_II_DEPRECATED` marker for now.

We've had this discussion before, but I want to once again point out the pain this causes in downstream projects. This change raises deprecation warnings from day one without the users having any time to use a different interface, so we force all downstream projects without very tight control on the deal.II version to be used `#ifdef` markers. For more commonly used functions like the present one, I suggest to have one release with both interfaces (and the preferred one in tutorials), one release with the deprecation warning, and then finally remove the old interface two releases ahead.

I want to contrast this to what I consider a gentle deprecation process, e.g. the `MatrixFree` class where we now deprecated `n_macro_cells` but we had at least one previous release with what we suggest as alternative. This still forces users to eventually upgrade and gives us a change to eventually fix an interface we do not like, but without being what I consider as explicitly rude against our users.

Reply:
> I agree with the general PR and with @deprecated in the doxygen comment, but I am very unhappy about adding the DEAL_II_DEPRECATED marker for now.